lib/handlers/injector-sor.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  CachingGasStationProvider,
+  CachingPoolProvider,
+  ChainId,
+  EIP1559GasPriceProvider,
+  ID_TO_NETWORK_NAME,
+  IGasPriceProvider,
+  IMetric,
+  IPoolProvider,
+  ISubgraphProvider,
+  ITokenListProvider,
+  ITokenProvider,
+  NodeJSCache,
+  PoolProvider,
+  QuoteProvider,
+  setGlobalLogger,
+  CachingTokenListProvider,
+  TokenProvider,
+  CachingTokenProviderWithFallback,
+  UniswapMulticallProvider,
+} from '@uniswap/smart-order-router';
+import { BaseRInj, Injector } from './handler';
+import {  ethers } from 'ethers';
+import { default as bunyan, default as Logger } from 'bunyan';
+import { Token } from '@uniswap/sdk-core';
+import NodeCache from 'node-cache';
+import { AWSTokenListProvider } from './router-entities/aws-token-list-provider';
+import { AWSSubgraphProvider } from './router-entities/aws-subgraph-provider';
+import { TokenList } from '@uniswap/token-lists';
+import UNSUPPORTED_TOKEN_LIST from './../config/unsupported.tokenlist.json';
+
+const SUPPORTED_CHAINS: ChainId[] = [ ChainId.MAINNET, ChainId.RINKEBY ];
+const DEFAULT_TOKEN_LIST = 'https://gateway.ipfs.io/ipns/tokens.uniswap.org';
+
+export interface RequestInjected<R> extends BaseRInj {
+  chainId: ChainId;
+  metric: IMetric;
+  poolProvider: IPoolProvider;
+  tokenProvider: ITokenProvider;
+  tokenListProvider: ITokenListProvider;
+  router: R;
+}
+
+export type ContainerDependencies = {
+  provider: ethers.providers.JsonRpcProvider;
+  subgraphProvider: ISubgraphProvider;
+  tokenListProvider: ITokenListProvider;
+  gasPriceProvider: IGasPriceProvider;
+  tokenProviderFromTokenList: ITokenProvider;
+  blockedTokenListProvider: ITokenListProvider;
+  poolProvider: IPoolProvider;
+  tokenProvider: ITokenProvider;
+  multicallProvider: UniswapMulticallProvider;
+  quoteProvider: QuoteProvider;
+};
+
+export interface ContainerInjected {
+  dependencies: {
+    [chainId in ChainId]?: ContainerDependencies;
+  }
+}
+
+export abstract class InjectorSOR<R, P> extends Injector<
```

Comment:
quote & quoteToRatio differ in Router and Params. I can extend these letters to signify these two things.